File: pinyougou_parent/pingyougou_sellergoods_interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbBrand;

import entity.PageResult;

import java.util.List;
import java.util.Map;

/**
 * 品牌接口
 */
public interface BrandService {
	/**
	 * 查询全表
	 * @return
	 * @throws Exception
	 */
    List<TbBrand> findAll();
    
    /**
     * 分页
     * @param pageNum
     * @param pageSize
     * @return
     */
    public PageResult findPage(int pageNum,int pageSize) ;
    
    
    // 增加一个品牌
    public void add(TbBrand brand) ;
    
    //根据id查询
    public TbBrand findOne(Long id) ;
    
    //修改
    public void update(TbBrand brand) ;
    
    //删除
    public void delete(Long[] ids) ;
    
    /**
	 * 条件查询分页
	 * @param pageNum 当前页面
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(TbBrand brand, int pageNum,int pageSize);
    
    
	/**
	* 品牌下拉框数据
	*/
	List<Map> selectOptionList();
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
}

File: pinyougou_parent/pingyougou_manager_web/src/main/webapp/js/service/brandService.js
// 品牌服务
		app.service("brandService",function($http){
			//查询全部
			this.findAll=function(){
				return $http.get('../brand/findAll.do');
			}
			// 分页
			this.findPage=function(page,size){
				return $http.get('../brand/findPage.do?page='+page +'&size='+size);
			}
			
			//根据id查询单个
			this.findOne=function(id){
				return $http.get('../brand/findOne.do?id='+id);
			}
			
			//添加方法
			this.add=function(entity){
				return $http.post('../brand/add.do',entity);
			}
			
			//修改方法
			this.update=function(entity){
				return $http.post('../brand/update.do',entity);
			}
			//删除方法
			this.dele=function(ids){
				return $http.get('../brand/delete.do?ids='+ids);
			}
			
			//条件分页查询
			this.search=function(page,size,searchEntity){
				return $http.post('../brand/search.do?page='+page +'&size='+size, searchEntity);
			}
			
			//下拉数据列表
			this.selectOptionList=function(){
				return $http.get('../brand/selectOptionList.do');
			}
			
		}); 